frontend-react/src/pages/submissions/SubmissionsTable.tsx

```diff
@@ -1,17 +1,78 @@
-import { useContext } from "react";
+import { useContext, useState } from "react";
 import moment from "moment";
 import { useResource } from "rest-hooks";
+import {
+    Button,
+    IconNavigateBefore,
+    IconNavigateNext,
+} from "@trussworks/react-uswds";
 
 import SubmissionsResource from "../../resources/SubmissionsResource";
 import { GlobalContext } from "../../components/GlobalContextProvider";
 
 function SubmissionsTable() {
     // this component will refresh when global context changes (e.g. organization changes)
     const globalState = useContext(GlobalContext);
+
+    // state of pagination
+    const [paginationCursor, setPaginationCursor] = useState("");
+    const [paginationSort, setPaginationSort] = useState("DESC");
+    const [paginationPageSize] = useState(10);
+
     const submissions: SubmissionsResource[] = useResource(
         SubmissionsResource.list(),
-        { organization: globalState.state.organization }
+        {
+            organization: globalState.state.organization,
+            cursor: paginationCursor,
+            pageSize: paginationPageSize,
+            sort: paginationSort,
+        }
     );
+
+    const sortedSubmissions = () => {
+        if (paginationSort === "ASC") {
+            // sort by createdAt DESC
+            // because when the paginationSort is ASC, the results from the server are reversed
+            submissions.sort(
+                (a, b) =>
+                    Date.parse(b.createdAt?.toString() || "") -
+                    Date.parse(a.createdAt?.toString() || "")
+            );
+        }
+        return submissions;
+    };
+
+    const updatePaginationCursor = (next: boolean) => {
+        let cursor = submissions[0]?.createdAt?.toString() || "";
+        let sort = "ASC";
+        if (next) {
+            cursor =
+                submissions[submissions.length - 1]?.createdAt?.toString() ||
+                "";
+            sort = "DESC";
+        }
+        setPaginationCursor(cursor);
+        setPaginationSort(sort);
+    };
+
+    // we can tell if we're on the first page by saving the first result and then checking against it later
+    const [firstPaginationCursor] = useState(
+        submissions[submissions.length - 1]?.createdAt?.toString() || ""
+    );
+
+    const onFirstPage = () => {
+        return submissions.find(
+            (s) => s.createdAt?.toString() === firstPaginationCursor
+        );
+    };
+
+    // 2022-01-07: the team suggested to see if the number of items matches
+    // the requested page size to tell if it is the last page,
+    // and then leave a message if there are no results on the next page
+    const onLastPage = () => {
+        return submissions.length !== paginationPageSize;
```

Comment:
So just to be sure I understand what we concluded, this will return `false` if the final page has the exact amount of submissions as the page size, thus showing the Next Page button and resulting in the "No results found" text on the last page, right?

I'm piecing together how this is all working, and I wonder if we could fix that experience for the users (I tend to dwell on things that are fun to solve). If we set the `pageSize` url parameter to `paginationPageSize + 1` and on response attempt to set a variable to `submissions[paginationPageSize] || null` (which should always return the last index because of 0 indexing), then we could make `onLastPage` return `firstSubmissionOnNextPage !== null` as our check. Might be a bit of over-engineering but I was feeling like thinking this out this morning. 😂